### DISCUSSION TITLE
Customise Arrow Hotkeys (Feature Request)

### DISCUSSION THREAD

**Discussion by Ahmed-Abdelgawad** [chrenn/swish-dl#16] [Q&A] [answered]
Hi, 
I would like to ask you if it is possible to remap the arrow Hotkeys to any other keys.  The reason is I want to map them to mouse keys. For example:
(Super modifier or Title bar) + Button 3 (Snap right)
(Super modifier or Title bar) + Button 4 (Snap left)
(Super modifier or Title bar) + one Scroll UP (Full Screen)
(Super modifier or Title bar) + two Scroll UP (Snap upper half)

The app is amazing, for trackpad features compared to every other window management it is snappier!

In my opinion, this simple customisation or feature would take it to the next level by bringing the mouse into the fold of Swish's intuitive interface.

I'm a big fan of Swish's capabilities with the trackpad and believe adding the ability to remap arrow hotkeys to mouse buttons would be a game-changer. This feature would make Swish a one-stop solution for both trackpad and mouse users, enhancing its usability and appeal.

I recently purchased a Magic Mouse to use with Swish, but found it uncomfortable, which highlighted the need for more mouse-based controls within the app. By allowing users to map functions to mouse buttons, Swish could cater to a wider audience and streamline workflows, especially for those who rely on or prefer using a mouse.

Thank you for considering this addition, which I believe would elevate Swish as the ultimate window management tool on macOS.

Best,

A. Abdelgawad

**Comment by chrenn** [comment, marked as answer]
Mouse keys and keyboard keys are fundamentally different, so remapping arrow hotkeys is not possible.

However, there's the fantastic [Mac Mouse Fix](https://github.com/noah-nuebling/mac-mouse-fix/releases), which emulates touch events for any third-party mouse. Just enable Smooth Scrolling or set any button's Click & Drag action to Scroll & Navigate.

I'm also working on some sort of mouse support for Swish V2.

**Comment by Ahmed-Abdelgawad** [comment]
Thanks for the reply and for suggesting Mac Mouse Fix, however it doesn't provide the missing feature from Swish. However, I'm excited to know that there will be some support for mouse in V2 👍
